pkg/kubelet/images/image_manager.go

```diff
@@ -164,6 +164,10 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 			msg := fmt.Sprintf("image pull failed for %s because the registry is unavailable.", container.Image)
 			return "", msg, imagePullResult.err
 		}
+		if imagePullResult.err.Error() == ErrInvalidSignature.Error() {
+			msg := fmt.Sprintf("image pull failed for %s because the signature is invalid.", container.Image)
+			return "", msg, imagePullResult.err
```

Comment:
should there be a difference in behavior? like should the kubelet stop trying to pull or something?

Reply:
That's a good question. Right now all CRI errors behave in the same way:

1. Initial pull via the CRI
2. If failed, show the error for 10s in the status, like:
    ```
    NAMESPACE     NAME                      READY   STATUS                RESTARTS   AGE
    default       test                      0/1     RegistryUnavailable   0          1s
    ```
3. On backoff time hit, show a new status:
    ```
    NAMESPACE     NAME                      READY   STATUS             RESTARTS   AGE
    default       test                      0/1     ImagePullBackOff   0          16s
    ```
4. After the backoff times out, we retry and show the desired error for another 10s:
    ```
    NAMESPACE     NAME                      READY   STATUS                RESTARTS   AGE
    default       test                      0/1     RegistryUnavailable   0          52s
    ```
5. …

The backoff time increases, means we mostly will show `ImagePullBackOff` to the end users and they have to check the events to get an idea about what is going on:

```
> k describe pod test | tail -n 4
  Warning  Failed     2m59s (x4 over 4m28s)  kubelet            Failed to pull image "localhost:5000/foo:1.0.0": RegistryUnavailable
  Warning  Failed     2m59s (x4 over 4m28s)  kubelet            Error: RegistryUnavailable
  Warning  Failed     2m47s (x6 over 4m27s)  kubelet            Error: ImagePullBackOff
  Normal   BackOff    2m35s (x7 over 4m27s)  kubelet            Back-off pulling image "localhost:5000/foo:1.0.0"
```

The backoff has no direct API to specify a time "unlimited" as far as I can see. Removing the entry within the backoff will simply re-pull every 10s. 

Means to exclude an image from pull, we would require something like an exclude map storing the last error (reason) why we do not re-pull it again. I can follow-up on that but it seems out of scope of this PR.

Reply:
you could imagine a transient signature check failure due to e.g. reporting this error when signature fails to fetch due to a network flake etc? I think it should probably keep retrying